core/client/fs/src/main/java/alluxio/client/file/FileSystemClientOptions.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0 (the
+ * "License"). You may not use this work except in compliance with the License, which is available
+ * at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file;
+
+import alluxio.Configuration;
+import alluxio.Constants;
+import alluxio.PropertyKey;
+import alluxio.client.ReadType;
+import alluxio.client.WriteType;
+import alluxio.grpc.CheckConsistencyPOptions;
+import alluxio.grpc.CompleteFilePOptions;
+import alluxio.grpc.CreateDirectoryPOptions;
+import alluxio.grpc.CreateFilePOptions;
+import alluxio.grpc.DeletePOptions;
+import alluxio.grpc.ExistsPOptions;
+import alluxio.grpc.FileSystemMasterCommonPOptions;
+import alluxio.grpc.FreePOptions;
+import alluxio.grpc.GetStatusPOptions;
+import alluxio.grpc.ListStatusPOptions;
+import alluxio.grpc.LoadMetadataPOptions;
+import alluxio.grpc.MountPOptions;
+import alluxio.grpc.OpenFilePOptions;
+import alluxio.grpc.ReadPType;
+import alluxio.grpc.RenamePOptions;
+import alluxio.grpc.SetAclPOptions;
+import alluxio.grpc.SetAttributePOptions;
+import alluxio.grpc.UfsPMode;
+import alluxio.grpc.UnmountPOptions;
+import alluxio.grpc.UpdateUfsModePOptions;
+import alluxio.grpc.WritePType;
+import alluxio.security.authorization.Mode;
+import alluxio.util.ModeUtils;
+import alluxio.util.grpc.GrpcUtils;
+import alluxio.wire.LoadMetadataType;
+import alluxio.wire.TtlAction;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * The file system class to set default options for client.
+ */
+@ThreadSafe
+public final class FileSystemClientOptions {
```

Comment:
Is this temporary? I thought we were going to fill in defaults on the server side

Reply:
Yes, but depends.
It's temporarily used here to provide client defaults. (Only the infra is done but the clients are still supplying defaults)

However, if there are discrepancies between client/master default values, then we would need to use this to merge client provided settings with client defaults, which would ultimately be merged with master defaults.

I'll tackle getting rid of client defaults as a separate work.